iotdb-core/datanode/src/main/java/org/apache/iotdb/db/service/DataNode.java

```diff
@@ -469,27 +469,45 @@ private void removeInvalidRegions(List<ConsensusGroupId> dataNodeConsensusGroupI
         DataRegionConsensusImpl.getInstance().getAllConsensusGroupIdsWithoutStarting().stream()
             .filter(consensusGroupId -> !dataNodeConsensusGroupIds.contains(consensusGroupId))
             .collect(Collectors.toList());
+    int length = invalidConsensusGroupIds.size();
+    invalidConsensusGroupIds.addAll(
```

Comment:
Break this into two functions, one to remove schemaregion and one to remove dataregion

Reply:
ok :)